h2o-mapreduce-generic/src/test/java/water/hadoop/CallbackManagerTest.java

```diff
@@ -0,0 +1,59 @@
+package water.hadoop;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import org.junit.rules.ExpectedException;
+import water.hadoop.h2odriver.CallbackManager;
+
+import java.net.Socket;
+import java.util.Collections;
+
+public class CallbackManagerTest {
+
+  @Rule
+  public ExpectedException ee = ExpectedException.none();
+  
+  @Test
+  public void testRegisterNode_retry() {
+    CallbackManager cm = new h2odriver().new CallbackManager(null, 5);
+
+    cm.registerNode("localhost", 54, 0, new Socket());
+    Socket s = new Socket();
+    cm.registerNode("localhost", 54, 1, s);
+    
+    assertEquals(cm._socks, Collections.singletonList(s));
+    assertEquals(cm._nodes, Collections.singletonList("localhost:54"));
+  }
+
+  @Test
+  public void testRegisterNode_outOfOrder() {
+    CallbackManager cm = new h2odriver().new CallbackManager(null, 5);
+
+    Socket s = new Socket();
+    cm.registerNode("localhost", 54, 1, s);
+    cm.registerNode("localhost", 54, 0, new Socket());
+
+    assertEquals(cm._socks, Collections.singletonList(s));
+    assertEquals(cm._nodes, Collections.singletonList("localhost:54"));
+  }
+
+  @Test
+  public void testRegisterNode_duplicate() {
```

Comment:
Should we add tests also for this case:

`Inconsistency found: old node entry for a repeated register node attempt doesn't exist` ?

Reply:
thanks, added